backend/LexBoxApi/GraphQL/OrgMutations.cs

```diff
@@ -89,6 +89,43 @@ public async Task<IQueryable<Organization>> AddProjectToOrg(
         return dbContext.Orgs.Where(o => o.Id == orgId);
     }
 
+    [Error<DbError>]
+    [Error<NotFoundException>]
+    [UseMutationConvention]
+    [UseFirstOrDefault]
+    [UseProjection]
+    public async Task<IQueryable<Organization>> AddProjectsToOrg(
+        LexBoxDbContext dbContext,
+        IPermissionService permissionService,
+        [Service] ProjectService projectService,
+        Guid orgId,
+        IEnumerable<Guid> projectIds)
+    {
+        var org = await dbContext.Orgs.Include(o => o.Members).Include(o => o.Projects).SingleOrDefaultAsync(o => o.Id == orgId);
+        NotFoundException.ThrowIfNull(org);
+        permissionService.AssertCanAddProjectToOrg(org);
+        // First make sure ALL projects pass permissions tests
+        foreach (var projectId in projectIds)
+        {
+            await permissionService.AssertCanManageProject(projectId);
+        }
+        // Now exclude any projects that don't actually exist or the org already has or that don't exist
+        var alreadyInOrg = dbContext.OrgProjects.Where(op => op.OrgId == orgId).Select(op => op.ProjectId).ToHashSet() ?? [];
+        var filteredIds = projectIds.Where(id => !alreadyInOrg.Contains(id));
+        // Also filter out any project IDs that don't really exist
+        var existingIds = await dbContext.Projects.Where(p => filteredIds.Contains(p.Id)).Select(p => p.Id).ToListAsync() ?? [];
```

Comment:
this won't filter anything as our permission check above will throw if the project does not exist

Reply:
Actually, if the logged-in user is an admin then AssertCanManageProject just returns true without doing any DB checks. Though the only way a logged-in admin could be submitting a request for a nonexistent project ID would be if they were submitting a manually-crafted GQL query, and I don't think that's going to happen. So sure, it's probably safe to remove this line and remove one DB query from the operation.

Reply:
Removed in commit 6681ba16.